### ISSUE TITLE
After a terraform apply, terraform plan says "silenced" changes need to be applied.

### ISSUE THREAD

**Issue by grayaii** [terraform-providers/terraform-provider-datadog#71]
### Terraform Version
Terraform v0.10.7

### Affected Resource(s)
datadog "silenced" map shows 

### Terraform Configuration Files
```hcl
resource "datadog_monitor" "prod_mem_high" {
  name               = "Memory usage is high"
  type               = "metric alert"
  message            = "@pagerduty-SRE"
  escalation_message = ""

  query = "avg(last_5m):avg:docker.mem.in_use{ecs_cluster:prod,!empire.app.name:graphql,!empire.app.name:mobile,!empire.app.name:graphql-beta} by {container_name} > ${var.prod_mem_high}"

  thresholds {
    critical          = "${var.prod_mem_high}"
  }

  notify_no_data      = "${var.notify_no_data}"
  renotify_interval   = "${var.renotify_interval}"
  new_host_delay      = "${var.new_host_delay}"
  require_full_window = "${var.require_full_window}"

  notify_audit = "${var.notify_audit}"
  timeout_h    = "${var.timeout_h}"
  locked       = "${var.locked}"

  no_data_timeframe = 2

  silenced {
    "*" = 1
  }

  tags = ["*"]
}
```

### Expected Behavior
After a terraform apply, there should be no changes in terraform plan.

### Actual Behavior
After a terraform apply, a terraform plan thinks there are changes to be applied:
~ datadog_monitor.prod_mem_high
       silenced.%: "0" => "1"
       silenced.*: "" => "1"

### Steps to Reproduce
1. `terraform apply`
2. `terraform plan`
You will see changes listed in "silenced"

### References
I see that this issue is closed:
https://github.com/terraform-providers/terraform-provider-datadog/issues/53
Which may be related to this.

**Comment by dariusjs** [comment]
I've also run into this with
```
Terraform v0.11.7
+ provider.datadog v1.0.3
```

**Comment by rkul** [comment]
Quick workaround is:
```hcl
lifecycle {
  ignore_changes = [
    "silenced"
  ]
}
```

**Comment by masci** [comment]
This seems related https://github.com/terraform-providers/terraform-provider-datadog/issues/41

**Comment by masci** [comment]
Closing as duplicate of #41, provided a fix through #125
